src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts

```diff
@@ -3358,5 +3378,71 @@ describe('chartOptionsBuilder', () => {
                 expect(chartOptions.yAxes).toEqual(expectedAxis);
             });
         });
+
+        describe('optional stacking', () => {
+            it('should return grouped categories with viewing by 2 attributes', () => {
+                const {
+                    data: {
+                        categories
+                    },
+                    isViewByTwoAttributes
+                } = generateChartOptions(fixtures.barChartWith4MetricsAndViewBy2Attribute);
+
+                expect(isViewByTwoAttributes).toBeTruthy();
+                expect(categories).toEqual([{
+                    name: 'Direct Sales',
+                    categories: ['East Coast', 'West Coast']
+                }, {
+                    name: 'Inside Sales',
+                    categories: ['East Coast', 'West Coast']
+                }]);
+            });
+
+            it('should not return grouped categories with viewing by one attribute', () => {
+                const {
+                    data: {
+                        categories
+                    },
+                    isViewByTwoAttributes
+                } = generateChartOptions(fixtures.barChartWith3MetricsAndViewByAttribute);
+
+                expect(isViewByTwoAttributes).toBeFalsy();
+                expect(categories).toEqual(['<button>2008</button>', '2009', '2010', '2011', '2012']);
+            });
+
+            it.each`
```

Comment:
This is really difficult to read. Most people don't even know it is possible. And using a template string table here seems like showing off. Leave it if you believe this is a good use case.

~~This is just painful to look at :-D~~ This doesn't communicate your intention clearly `${'undefined'} | ${undefined}`

Reply:
Hi @tomasreichmann ,
We can follow `it.each([...])('should %s', (_desc: string, b, c) => {...})`
But we will have unused placeholder `_desc`, I see it not really nice.
If `b, c` are simple params, I think we can use string template to avoid using placehilder `_desc`.

About `${undefined}`, I think the description `should return 'undefined' stacking with stack options are $description` tell you what I want :)